crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -417,6 +417,70 @@ impl WaylandWindowStatePtr {
         }
     }
 
+    pub fn handle_ime_commit(&self, text: String) {
+        if let Some(ref mut fun) = self.callbacks.borrow_mut().input {
+            if !fun(PlatformInput::KeyUp(crate::KeyUpEvent {
+                keystroke: crate::Keystroke::default(),
+            }))
+            .propagate
+            {
+                return;
+            }
+        }
+        let mut state = self.state.borrow_mut();
+        if let Some(mut input_handler) = state.input_handler.take() {
+            drop(state);
+            input_handler.replace_text_in_range(None, &text);
+            self.state.borrow_mut().input_handler = Some(input_handler);
+        }
+    }
+
+    pub fn handle_ime_preedit(&self, text: String) {
+        if let Some(ref mut fun) = self.callbacks.borrow_mut().input {
+            if !fun(PlatformInput::KeyUp(crate::KeyUpEvent {
+                keystroke: crate::Keystroke::default(),
+            }))
+            .propagate
+            {
+                return;
+            }
+        }
+        let mut state = self.state.borrow_mut();
+        if let Some(mut input_handler) = state.input_handler.take() {
+            drop(state);
+            input_handler.replace_and_mark_text_in_range(
+                None,
+                &text,
+                Some(0..text.chars().count()),
```

Comment:
I believe the third argument should be `None`. Even if it's not set an underline ~~is~~ **should** be rendered (it isn't right now, but it's an unrelated bug).

See https://github.com/zed-industries/zed/pull/7488

P.S: Selecting text also copies it to the primary clipboard, which is also not desired.